src/main/java/seedu/cafectrl/storage/Decoder.java

```diff
@@ -166,45 +167,114 @@ public static Sales decodeSales(ArrayList<String> textLines, Menu menu) {
+        try {
+            String[] orderData = orderLine.split(DIVIDER);
+            int day = Integer.parseInt(orderData[0].trim()) - Sales.DAY_DISPLAY_OFFSET;
+            String dishName = orderData[1].trim();
+
+            //@@author Cazh1
+            //keeps track of the number of days cafe has been operating for
+            if (dishName.equals(Encoder.NULL_ORDER_DAY)) {
+                fillOrderListSize(orderLists, day);
+                return;
+            }
+            //@@author
+
+            int quantity = Integer.parseInt(orderData[2].trim());
+            float decodedTotalOrderCost = Float.parseFloat(orderData[3].trim());
+            String completeStatus = orderData[4].trim();
+            Dish dish = menu.getDishFromName(dishName);
+
+            boolean isDataAccurate = isDishValid(orderLine, dish)
+                    && isOrderCostAccurate(orderLine, dish, quantity, decodedTotalOrderCost)
+                    && isCompleteStatusAccurate(orderLine, completeStatus);
+            if (!isDataAccurate) {
+                return;
+            }
+
+            //creates new order and adds to orderList for specific day
+            boolean isComplete = Boolean.parseBoolean(completeStatus.toLowerCase());
+            Order orderedDish = new Order(menu.getDishFromName(dishName), quantity, decodedTotalOrderCost, isComplete);
+            fillOrderListSize(orderLists, day);
+            orderLists.get(day).addOrder(orderedDish);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Line corrupted: " + e.getMessage(), e);
+            ui.showToUser(ErrorMessages.INVALID_SALES_DATA + orderLine);
+        }
+    }
+
+    /**
+     * Checks if the Dish is valid (exists in menu) and shows an error message if it's not.
+     *
+     * @param orderLine The order line in the format "day|dishName|quantity|totalOrderCost|isComplete".
+     * @param dish      The Dish object to be validated.
+     * @return True if the Dish is valid, false otherwise.
+     */
+    private static boolean isDishValid(String orderLine, Dish dish) {
+        if (dish != null) {
+            return true;
+        }
+        ui.showToUser(ErrorMessages.INVALID_ORDER_DATA + orderLine);
+        return false;
+    }
+
+    /**
+     * Checks if the decoded total order cost matches the expected total order cost and shows an error message if not.
+     *
+     * @param orderLine            The order line in the format "day|dishName|quantity|totalOrderCost|isComplete".
+     * @param dish                 The Dish object in the order.
+     * @param quantity             The quantity of the dish in the order.
+     * @param decodedTotalOrderCost The decoded total order cost from the order line.
+     * @return True if the decoded total order cost matches the expected total order cost, false otherwise.
+     */
+    private static boolean isOrderCostAccurate(String orderLine, Dish dish,
```

Comment:
Perhaps the logic behind this method would cause logical flaws.
If an order was completed yesterday at $2.50, any changes to the price afterwards would cause the already completed order to be flagged as invalid as the price then would not match the most updated price

Reply:
Thank you for highlighting this feature flaw! I have made the necessary changes!